sig/datadog/core/diagnostics/environment_logger.rbs

```diff
@@ -14,7 +14,7 @@ module Datadog
       module EnvironmentLogger
         extend EnvironmentLogging
 
-        def self.collect_and_log!: (Hash[Symbol, untyped] extra_fields) -> void
+        def self.collect_and_log!: (?Hash[Symbol, untyped]? extra_fields) -> void
```

Comment:
Minor: I checked the code + tested locally, this can be a bit stricter

```suggestion
        def self.collect_and_log!: (Hash[Symbol, untyped]? extra_fields) -> void
```